### ISSUE TITLE
indicate different creators/narrators per resource/track

### ISSUE THREAD

**Issue by rakutenjeff** [w3c/wpub#229]
For example conference publication as one WP each article in the proceedings could be written by different creators or for audio book each chapter could be read by different narrators.
We also need to consider not to make manifest for each resource too complex though.

**Comment by HadrienGardeur** [comment]
I think that there are potentially two different approach to this issue:

1. delegate all resource-level metadata to the resource itself
2. extend our current model to also support such metadata in `PublicationLink`

For HTML resources, this group has a strong preference for option 1, but with images/audio/video things are quite different and there's a wide variety of formats to consider.

If we go down the road of option 2, this would look something like this:
```
{
  "url": "chapter1.mp3",
  "encodingFormat": "audio/mpeg",
  "readBy": "John Doe"
}
```

Based on our current draft, this would probably not end up in our WebIDL but it wouldn't get rejected by our JSON Schema either.

**Comment by HadrienGardeur** [comment]
For our next call, we need to decide if this is something that we want to be able to express in our manifest.

**Comment by rakutenjeff** [comment]
I start to feel this might need to be resolved in more wide WP scope since for example Academic Journal each article (chapter) might have different creator.
Audio book might have difference comparing with WP that one mp3 file might involve different narrators but who would create this type of audio book and say "Hi we need to distinguish narrator for different track".

The json in https://github.com/w3c/wpub/issues/229#issuecomment-413520551 is a good example

**Comment by HadrienGardeur** [comment]
@rakutenjeff I disagree and don't think we should change the scope of this discussion.

In the case of a WP where `readingOrder` points to HTML resources, we can always include in that resource:

- JSON-LD in the resource itself in a `script` element
- RDFa
- or microdata

With audio and video, there are multiple formats and potentially different metadata for these different formats as well. That's the reason why the situation is a little bit different.

**Comment by rakutenjeff** [comment]
yes thinking about handling different format of audio/video with different metadata this would be different from general WP. I will change title to be more specifically.

**Comment by rakutenjeff** [comment]
Feel free to make the title to be more accurate @HadrienGardeur .

**Comment by rakutenjeff** [comment]
Handle different format might be more important for RS which can support audio book.

Say we have following two format for "same" resource and mp3 is fallback of ogg:
```
{
  "url": "chapter1.mp3",
  "encodingFormat": "audio/mpeg",
  "readBy": "John Doe"
}
{
  "url": "chapter1.ogg",
  "encodingFormat": "audio/ogg",
  "readBy": "John Doe"
}
```
How do we express this in both option #1 and #2.

**Comment by HadrienGardeur** [comment]
Going back to the two options listed in https://github.com/w3c/wpub/issues/229#issuecomment-413520551:

> 1. delegate all resource-level metadata to the resource itself

This might be problematic if we need to display such information without downloading all the audio tracks first.

Would the list of tracks ever get displayed to the user in an audiobook? If so, would the UI need to display the narrator and/or author in addition to `name`/`description` (which we already have)?

> 2. extend our current model to also support such metadata in `PublicationLink`

This would need to be addressed in the spec. Some of those terms are only allowed in `CreativeWork` or `Audiobook` right now.

WP might switch from its own model (`PublicationLink`) to an alternative that's part of schema.org. In this case, we would lose some flexibility in the process.

**Comment by HadrienGardeur** [comment]
> Handle different format might be more important for RS which can support audio book.
>
> Say we have following two format for "same" resource and mp3 is fallback of ogg:

That's a different issue, see https://github.com/w3c/wpub/issues/308

**Comment by rakutenjeff** [comment]
We have a vote [here](https://www.w3.org/2018/08/28-pwg-audio-minutes.html) of choosing publication level of description of resource level (see following). `publication level` is the winner.

```
duga: trying to impart meaning on arbitrary media resources is the wrong direction
... if we want to really do this, we need to point to time segments, not resources
... resource files don't have inherent meaning

wendyreid: there are two options
... high-level metadata, everyone gets listed at the top level
... for other stuff you can use description
... or we can go to track/time level

Hadrien: I want a vote
... I think description enough
... brady is right htat how the content is divided is meaningless
... but timestamps are tricky, if it crosses boundary of multiple resources
... for now I think description is enough

<George> It seems that the boundry of specfic speakers or narrators is more in the domain of synced media, the community Group Marisa leads.

dkaplan3: I'm good with a vote, and with the less-granular version
... there will be a long tail of things we wish audio books could do that they can't do
... a lot of publishers will not implement anyway
... the perfect is the enemy of the good, as Ivan said the other day

zheng_xu: I think this depends on how we define duration
... then how we define duration for each narrator
... not possible to generate automatically, so that's why we need info in manifest

wendyreid: let us have the vote
... do we want to say that narrator/creator info be in publication-level metadata, or resource-level metadata?
```

**Comment by HadrienGardeur** [comment]
Just to illustrate what's possible with `name` + `description`:

```
{
  "url": "chapter1.mp3",
  "encodingFormat": "audio/mpeg",
  "name": "Chapter 1",
  "description": "Narrated by Joe Narrator"
}
```

As pointed out by @bdugas though, how the publication is divided into multiple audio resources is completely arbitrary. 
If we wanted to truly support the expression of a narrator for a sub-set of a publication, we would first need the ability to scope things across:

- one or more resources
- specific fragments of such resources (possible through the use of media fragments)

This gets complex pretty quickly and seems to be more in the realm of synchronized media than this WG or TF.

**Comment by HadrienGardeur** [comment]
Since we've had a vote on this issue, I propose closing now.
